proposals/0000-proposal-process.rst

```diff
@@ -0,0 +1,319 @@
+**This proposal is open for community comments!** Please let us know what you
+think about the proposed protocol by leaving a comment on the GitHub pull
+request associated with the proposal by clicking on the "Conversation" tab above.
+
+A Proposal for Proposals
+========================
+
+.. contents::
+
+Recently there has been a growing feeling within the Haskell community that the
+mechanisms for proposing features and changes to GHC and the language that it
+compiles are less than approachable. In particular our tools for managing the
+proposal specifications and surrounding discussions make participation harder
+than necessary. Over the last few weeks I have been looking at how other
+open-source compiler projects address these issues (focusing on LLVM, Python,
+and Rust) with an eye towards improving GHC's own practices.
+
+To be clear, the process discussed below only affects proposal of the following
+three classes,
+
+* A syntactic change to GHC Haskell (e.g. the various ``ShortImports``
+  `proposals <https://ghc.haskell.org/trac/ghc/ticket/10478>`_, ``do``
+  `expressions <https://ghc.haskell.org/trac/ghc/ticket/10843>`_ without ``$``)
+
+* A major change to the user-visible behaviour of the compiler (e.g. the recent
+  `change <https://ghc.haskell.org/trac/ghc/ticket/11762>`_ in super-class
+  solving, and ``-Wall`` `behavior <https://ghc.haskell.org/trac/ghc/ticket/11370>`_)
+
+* The addition of major features to the compiler (e.g. ``-XTypeInType``, GHCi
+  `commands <https://ghc.haskell.org/trac/ghc/ticket/10874>`_,
+  `type-indexed <https://ghc.haskell.org/trac/ghc/wiki/Typeable>`_
+  ``Typeable`` representations)
+
+* The removal of major features from the compiler
```

Comment:
We could list some classes which clearly falls out of the scope of this process.

Rust lists e.g.
- *Additions only likely to be noticed by other developers-of-rust, invisible to users-of-rust.*
- *Additions that strictly improve objective, numerical quality criteria (warning removal, speedup, better platform coverage, more parallelism, trap more errors, etc.)* (I can think e.g. about GADT pattern matching exhaustiveness checker)